pennylane/gradients/vjp.py

```diff
@@ -47,7 +47,7 @@ def _all_close_to_zero(dy):
         return qml.math.allclose(dy, 0)
 
     # call this method recursively
-    return qml.math.all(qml.math.stack([_all_close_to_zero(dy_) for dy_ in dy]))
+    return all(_all_close_to_zero(dy_) for dy_ in dy)

```

Comment:
Two questions here:
- Could you explain what is happening with this line change?
- It appears we still use `_all_close_to_zero` early on in `vjp` - why is it ok there, and why did we also have it in `compute_vjp_single`?

Reply:
Since this is returning a true/false only, we don't need to stack etc on the returned output. Also `all` with a generator exits early, so it will stop evaluating the outputs if we get a non close to zero component, reducing the runtime overall.

Keeping this for `vjp` allows us to preserve the current behaviour of reduced executions for hardware devices.

Reply:
> Keeping this for vjp allows us to preserve the current behaviour of reduced executions for hardware devices.

Not a blocker for merging, but I'm still confused here. It seems like `vjp` will check for `dy` being close to zero multiple times - once at this point and once again in the `processing_fn`. The latter has been disabled in this PR, but I don't understand why it was needed in the first place if we're doing the check here. Is it a subtlety with forward and backward passes?